python/revng/daemon/event_manager.py

```diff
@@ -0,0 +1,135 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import logging
+import os
+import re
+import signal
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from concurrent.futures import Future, ThreadPoolExecutor
+from enum import Enum, auto
+from functools import wraps
+from pathlib import Path
+from tempfile import mkdtemp
+from typing import Dict, List, Protocol
+
+from starlette.requests import Headers
+
+from revng.api import Manager
+
+
+class EventType(Enum):
+    BEGIN = auto()
+    CONTEXT = auto()
+
+
+def emit_event(event_type: EventType):
```

Comment:
This is not straightforward, it does some magic with the first argument.
Docs!

Reply:
Added